benchmarks/datasets/wikitext2_data.py

```diff
@@ -52,30 +54,58 @@ def data_process(raw_text_iter):
     train_dataset = data_process(iter(io.open(train_filepath, encoding="utf8")))
     valid_dataset = data_process(iter(io.open(valid_filepath, encoding="utf8")))
     test_dataset = data_process(iter(io.open(test_filepath, encoding="utf8")))
+    return DatasetsInfo(len(vocab.stoi), train_dataset, valid_dataset, test_dataset)
+
 
+def get_dataloaders(datasets_info, benchmark_config, model_specs, num_replicas=1, rank=0):
+    ntokens, train_dataset, valid_dataset, test_dataset = datasets_info
     def batchify(data):
         batch_size = benchmark_config["batch_size"]
         return _batchify(data, batch_size)
 
     total_batch_size = _get_total_batch_size(benchmark_config, model_specs)
-    train_dataloader = DataLoader(train_dataset, batch_size=total_batch_size, collate_fn=batchify)
-    valid_dataloader = DataLoader(valid_dataset, batch_size=total_batch_size, collate_fn=batchify)
-    test_dataloader = DataLoader(test_dataset, batch_size=total_batch_size, collate_fn=batchify)
-    return len(vocab.stoi), train_dataloader, valid_dataloader, test_dataloader
-
+    train_dataloader = DataLoader(
+            train_dataset,
+            sampler=DistributedSampler(train_dataset, num_replicas=num_replicas, rank=rank),
+            batch_size=total_batch_size,
+            collate_fn=batchify)
+    valid_dataloader = DataLoader(
+            valid_dataset,
+            sampler=DistributedSampler(valid_dataset, num_replicas=num_replicas, rank=rank),
+            batch_size=total_batch_size,
+            collate_fn=batchify)
+    test_dataloader = DataLoader(
+            test_dataset,
+            sampler=DistributedSampler(test_dataset, num_replicas=num_replicas, rank=rank),
+            batch_size=total_batch_size,
+            collate_fn=batchify)
+    return train_dataloader, valid_dataloader, test_dataloader
+
+
+
+def get_real_dataloaders(args, benchmark_config, model_specs, num_replicas=1, rank=0):
+    """Return real dataloaders for training, testing and validation."""
+    dataset_info = get_real_datasets()
+    x, y, z = get_dataloaders(
```

Comment:
Lets actual use the "train, test and valid dataloader" names here for clarity.

Reply:
done